pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -539,6 +542,13 @@ private List<PinotTaskConfig> createPinotTaskConfigs(List<SegmentZKMetadata> sel
     }
 
     List<PinotTaskConfig> pinotTaskConfigs = new ArrayList<>();
+    StringBuilder partitionSuffixBuilder = new StringBuilder();
+    if (partition != null && !partition.isEmpty()) {
+      for (int columnPartition : partition) {
+        partitionSuffixBuilder.append(DELIMITER_IN_SEGMENT_NAME).append(columnPartition);
```

Comment:
This can potentially result in a very long segment name if the upstream user doesn't actually partition properly. We have added `System.currentTimeMillis()` and `i` to avoid the segment name issue in general. 

```
e.g. Kafka with 100 partition

segment_name_0_1_2_3_4...._100_...
``` 
@jtao15 Can you double check on this?

Reply:
That is a great point!

As discussed offline, instead of appending partition info directly, we can map each to a unique sequence number.

Reply:
Discussed offline again, we decided to change back to the initial solution. This will not be a problem due to line 375~376.